client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
+		const post = getSitePost( getState(), action.siteId, action.postId );
+
+		let message;
+		if ( post ) {
+			message = translate( 'An error occurred while deleting "%s"', {
+				args: [ truncate( post.title, { length: 24 } ) ]
+			} );
+		} else {
+			message = translate( 'An error occurred while deleting the post' );
+		}
+
+		dispatch( errorNotice( message ) );
+	},
+	[ POST_DELETE_SUCCESS ]: ( action, dispatch ) => {
+		dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
+	},
+	[ POST_SAVE_SUCCESS ]: ( action, dispatch ) => {
+		let text;
+		switch ( action.post.status ) {
+			case 'trash':
+				text = translate( 'Post successfully moved to trash' );
+				break;
+
+			case 'publish':
+				text = translate( 'Post successfully published' );
+				break;
+		}
+
+		if ( text ) {
+			dispatch( successNotice( text ) );
+		}
+	}
+};
+
+export default function noticesMiddleware( { dispatch, getState } ) {
+	return ( next ) => ( action ) => {
```

Comment:
I'm guessing you prefer to not do this, but many of the Redux libraries tend to use fat-arrow functions and without parenthesis for single argument functions (not sure if this is experience or my personal bias giving me this impression). to me it looks far better than the verbose mixture of `function( … ) { return ( next ) => ( action ) => … }`

```js
export const noticesMiddleware = ( { dispatch, getState } ) => next => action => {
	…
}
```

also, this threw me for a loop, because I see different definitions for the middleware API depending on where I look. turns out this has been a controversial thing and although I found several PRs/issues dealing with changing the signature of `applyMiddleware`, I never found the commits that actually changed it `¯\_(ツ)_/¯`